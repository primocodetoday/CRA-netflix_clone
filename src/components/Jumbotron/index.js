import React from 'react';
import PropTypes from 'prop-types';
import {
  Item,
  Inner,
  Container,
  Pane,
  Title,
  SubTitle,
  Image,
} from './styles/jumbotron';

const Jumbotron = ({ children, direction }, ...restProps) => {
  return (
    <Item {...restProps} direction={direction}>
      <Inner>{children}</Inner>
    </Item>
  );
};

Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane({ children, ...restProps }) {
  return <Pane {...restProps}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.SubTitle = function JumboSubtronTitle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Jumbotron.Image = function JumbotronImage({ ...restProps }) {
  return <Image {...restProps} />;
};

Jumbotron.propTypes = {
  direction: PropTypes.string,
};

Jumbotron.defaultProps = {
  direction: 'row',
};

export default Jumbotron;
