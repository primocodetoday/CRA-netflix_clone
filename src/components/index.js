export { default as Footer } from 'components/Footer';
export { default as Accordion } from 'components/Accordion';
export { default as Jumbotron } from 'components/Jumbotron';
