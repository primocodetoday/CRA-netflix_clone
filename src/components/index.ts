export { Accordion } from 'components/Accordion/Accordion';
export { Banner } from 'components/Banner/Banner';
export { Card } from 'components/Card/Card';
export { CTAForm } from 'components/CTAForm/CTAForm';
export { Form } from 'components/Form/Form';
export { Footer } from 'components/Footer/Footer';
export { Header } from 'components/Header/Header';
export { Jumbotron } from 'components/Jumbotron/Jumbotron';
export { Loading } from 'components/Loading/Loading';
export { Player } from 'components/Player/Player';
export { Profiles } from 'components/Profiles/Profiles';
