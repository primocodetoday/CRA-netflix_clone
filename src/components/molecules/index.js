export { default as Footer } from 'components/molecules/Footer/Footer';
export { default as Accordion } from 'components/molecules/Accordion/Accordion';
export { default as Jumbotron } from 'components/molecules/Jumbotron/Jumbotron';
export { default as CTAForm } from 'components/molecules/CTAForm/CTAForm';
export { default as Header } from 'components/molecules/Header/Header';
