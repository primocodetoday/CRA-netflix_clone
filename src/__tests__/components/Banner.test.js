import React from 'react';

import { Banner } from 'components';
import { render } from '@testing-library/react';

describe('<Banner/> render properly', () => {
  it('with sub-components & children', () => {
    const { container, getByText } = render(
      <Banner>
        <Banner.Title>Unlimited movies</Banner.Title>
        <Banner.SubTitle>Watch anywhere</Banner.SubTitle>
      </Banner>,
    );

    expect(getByText('Unlimited movies')).toBeTruthy();
    expect(getByText('Watch anywhere')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('with just a title', () => {
    const { container, queryByText, getByText } = render(
      <Banner>
        <Banner.Title>Unlimited films</Banner.Title>
      </Banner>,
    );

    expect(getByText('Unlimited films')).toBeTruthy();
    expect(queryByText('Watch anywhere')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('with just a sub-title', () => {
    const { container, queryByText, getByText } = render(
      <Banner>
        <Banner.SubTitle>Watch anywhere</Banner.SubTitle>
      </Banner>,
    );

    expect(getByText('Watch anywhere')).toBeTruthy();
    expect(queryByText('Unlimited films')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
