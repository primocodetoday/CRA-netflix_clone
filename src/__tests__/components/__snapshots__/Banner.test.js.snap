// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner/> render properly with just a sub-title 1`] = `
<div
  class="StyledBanner__Container-j7lj91-0 eYceKL"
>
  <h2
    class="StyledBanner__SubTitle-j7lj91-2 fOkbqo"
  >
    Watch anywhere
  </h2>
</div>
`;

exports[`<Banner/> render properly with just a title 1`] = `
<div
  class="StyledBanner__Container-j7lj91-0 eYceKL"
>
  <h1
    class="StyledBanner__Title-j7lj91-1 kDIfLf"
  >
    Unlimited films
  </h1>
</div>
`;

exports[`<Banner/> render properly with sub-components & children 1`] = `
<div
  class="StyledBanner__Container-j7lj91-0 eYceKL"
>
  <h1
    class="StyledBanner__Title-j7lj91-1 kDIfLf"
  >
    Unlimited movies
  </h1>
  <h2
    class="StyledBanner__SubTitle-j7lj91-2 fOkbqo"
  >
    Watch anywhere
  </h2>
</div>
`;
